app/lib/experiments.coffee

```diff
@@ -9,7 +10,7 @@ AnalyticsLogger = require 'lib/analytics'
 Define the active experiment here by using a string which exists in http://experiments.zooniverse.org/active_experiments
 If no experiments should be running right now, set this to null, false or ""
 ###
-ACTIVE_EXPERIMENT = null
+ACTIVE_EXPERIMENT = "SerengetiBlanksExperiment1"
```

Comment:
You're partly correct.

At the moment, participation in an experiment is not user-specific, it's either all or none, so it doesn't necessarily make sense or offer a benefit to move that setting to the experiment server. However, I am not against doing so, and I guess it's slightly tidier. I will add an issue to Exp server to do this improvement.

However, having custom code that is aware of the experiment is still inevitable, for the time being, as the decisions to be made about subject selection are specific to the experiment. This can be improved though.

For the last part, yes this is absolutely how we get the cohort.

But not the source. Source is a property of a subject, not of a user. At the moment that is passed over as part of the nextSubjectID. Not ideal, but it works. I will probably improve this later.